Marlin/ultralcd.cpp

```diff
@@ -2232,8 +2234,16 @@ void kill_screen(const char* lcd_msg) {
 
         ubl_map_move_to_xy(); // Move to current location
 
-        if (planner.movesplanned() > 1) { // if the nozzle is moving, cancel the move. There is a new location
-          quickstop_stepper();
+        if (planner.movesplanned() > 1) { // if the nozzle is moving, cancel the move.  There is a new location
+          #define ENABLE_STEPPER_DRIVER_INTERRUPT()  SBI(TIMSK1, OCIE1A)
+          #define DISABLE_STEPPER_DRIVER_INTERRUPT() CBI(TIMSK1, OCIE1A)
+          DISABLE_STEPPER_DRIVER_INTERRUPT();
+          while (planner.blocks_queued()) planner.discard_current_block();
+          stepper.current_block = NULL;
+          planner.clear_block_buffer_runtime();
+          ENABLE_STEPPER_DRIVER_INTERRUPT();
+          set_current_from_steppers_for_axis(ALL_AXES);
+          sync_plan_position();
```

Comment:
Rather than duplicate all of what `quickstop_stepper` does here, can you add a boolean argument to `quickstop_stepper` and have it modify its behavior for what you need? This code "knows too much" about the inner workings of the stepper code, and should not be here.